crates/distribution-types/src/installed.rs

```diff
@@ -32,6 +33,7 @@ pub struct InstalledRegistryDist {
 pub struct InstalledDirectUrlDist {
     pub name: PackageName,
     pub version: Version,
+    pub direct_url: Box<DirectUrl>,
```

Comment:
Why boxed?

Reply:
Clippy complains that the `InstalledDist` enum has a large variant size difference otherwise due to the large `DirectUrl` type, and boxing here is minimally intrusive.

```
error: large size difference between variants
  --> crates/distribution-types/src/installed.rs:18:1
   |
18 | / pub enum InstalledDist {
19 | |     /// The distribution was derived from a registry, like `PyPI`.
20 | |     Registry(InstalledRegistryDist),
   | |     ------------------------------- the second-largest variant contains at least 56 bytes
21 | |     /// The distribution was derived from an arbitrary URL.
22 | |     Url(InstalledDirectUrlDist),
   | |     --------------------------- the largest variant contains at least 272 bytes
23 | | }
   | |_^ the entire enum is at least 272 bytes
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant
   = note: `-D clippy::large-enum-variant` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::large_enum_variant)]`
help: consider boxing the large fields to reduce the total size of the enum
   |
22 |     Url(Box<InstalledDirectUrlDist>),
   |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```